runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -626,52 +628,119 @@ public synchronized long find(long globalAddress, SearchDirection direction) {
+                    latestValidCheckpoint.addAddress(data.getGlobalAddress());
+                }
+            } else if (latestValidCheckpoint.getCheckpointId() != null &&
+                    latestValidCheckpoint.getCheckpointId().equals(cpEntry.getCheckpointId())) {
+                // Case: all other markers other than END of a checkpoint.
+
+                // Add checkpoint entry data to the summarized state of the checkpoint, which will be used
+                // when the definite checkpoint is selected.
+                latestValidCheckpoint.addBytes((long) data.getSizeEstimate());
+                latestValidCheckpoint.addNumEntries(1);
+                latestValidCheckpoint.addAddress(data.getGlobalAddress());
 
-                context.checkpointSuccessNumEntries = 1L;
-                context.checkpointSuccessBytes = (long) data.getSizeEstimate();
-                context.checkpointSuccessEndAddr = data.getGlobalAddress();
-            }
-            else if (data.getCheckpointId().equals(context.checkpointSuccessId)) {
-                context.checkpointSuccessNumEntries++;
-                context.checkpointSuccessBytes += cpEntry.getSmrEntriesBytes();
                 if (cpEntry.getCpType().equals(CheckpointEntry.CheckpointEntryType.START)) {
-                    context.checkpointSuccessStartAddr = Long.decode(cpEntry.getDict()
-                            .get(CheckpointEntry.CheckpointDictKey.START_LOG_ADDRESS));
-                    if (cpEntry.getDict().get(CheckpointEntry.CheckpointDictKey
-                            .SNAPSHOT_ADDRESS) != null) {
-                        context.checkpointSnapshotAddress = Long.decode(cpEntry.getDict()
-                                .get(CheckpointEntry.CheckpointDictKey.SNAPSHOT_ADDRESS));
-                    }
-                    log.trace("Checkpoint[{}] HALT due to START at address {} startAddr"
-                                    + " {} type {} id {} author {}",
-                            this, data.getGlobalAddress(), context.checkpointSuccessStartAddr,
-                            cpEntry.getCpType(),
+                    // Only for the case of START markers add some extra information.
+                    log.trace("Checkpoint[{}] START found at address {} type {} id {} author {}",
+                            this, data.getGlobalAddress(), cpEntry.getCpType(),
                             Utils.toReadableId(cpEntry.getCheckpointId()),
                             cpEntry.getCheckpointAuthorId());
-                    return BackpointerOp.INCLUDE_STOP;
+
+                    if (cpEntry.getDict().get(CheckpointEntry.CheckpointDictKey
+                                        .SNAPSHOT_ADDRESS) != null) {
+                        latestValidCheckpoint.setCheckpointSnapshotAddress(Long.decode(cpEntry.getDict()
+                                .get(CheckpointEntry.CheckpointDictKey.SNAPSHOT_ADDRESS)));
```

Comment:
On Friday we checked each checkpoint entry contains the checkpoint metadata SNAPSHOT_ADDRESS, so we can setCheckpointSnapshotAddress  when reading the END checkpoint entry and remove this block, right?

Reply:
Right! Removed.